util/versions/version_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package versions
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGetHigherK8sVersion(t *testing.T) {
+	cases := []struct {
+		name   string
+		a      string
+		b      string
+		output string
+	}{
+		{
+			name:   "a is greater than b",
+			a:      "v1.17.8",
+			b:      "v1.18.8",
+			output: "v1.18.8",
+		},
+		{
+			name:   "b is greater than a",
+			a:      "v1.18.9",
+			b:      "v1.18.8",
+			output: "v1.18.9",
+		},
+		{
+			name:   "b is greater than a",
+			a:      "v1.18",
+			b:      "v1.18.8",
+			output: "v1.18.8",
+		},
+		{
+			name:   "a is equal to b",
+			a:      "v1.18.8",
+			b:      "v1.18.8",
+			output: "v1.18.8",
+		},
+		{
+			name:   "a is greater than b and a is major.minor",
+			a:      "v1.18",
+			b:      "v1.17.8",
+			output: "v1.18",
+		},
+		{
+			name:   "a is greater than b and a is major.minor",
+			a:      "1.18",
+			b:      "1.17.8",
+			output: "1.18",
+		},
+		{
+			name:   "a is invalid",
+			a:      "1.18.",
+			b:      "v1.17.8",
+			output: "v1.17.8",
+		},
+		{
+			name:   "b is invalid",
+			a:      "1.18.1",
+			b:      "v1.17.8.",
+			output: "1.18.1",
+		},
+		{
+			name:   "b is invalid",
+			a:      "9.99.9999",
+			b:      "v1.17.8.",
+			output: "9.99.9999",
```

Comment:
Just curious, although this unit test tests the function, in real life scenario, is `9.99.9999` a meaningful SemVer? 
Can user replicate this scenario?

Reply:
Not sure, was just trying out different test cases and checking if the function works fine that's is :slightly_smiling_face: